### ISSUE TITLE
Code compiled with GCC 10.2.0_2 seems to unable to emit exception code.

### ISSUE THREAD

**Issue by hanickadot** [Homebrew/homebrew-core#69614]
# Bug report

I updated GCC build to latest one, and noticed my code started crashing, after some investigation I found out it's happening with latest version of GCC 10.2.0_2 and it didn't on 10.2.0. I was able to confirm that on second machine.

- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

```
 ~ $ brew config
HOMEBREW_VERSION: 2.7.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: bca4804a9e48de5319383d3eddadaa7f054c77da
Last commit: 8 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 2440c2d42cf1f0f75e4c0f3c01c37c80130a7e85
Core tap last commit: 53 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit icelake
Clang: 12.0 build 1200
Git: 2.30.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.1-x86_64
CLT: 12.3.0.0.1.1607026830
Xcode: 12.3
 ~ $ brew doctor
Your system is ready to brew.
```

## What you were trying to do (and why)

Updated GCC and tried to compile my code as usual.

## What happened (include command output)

The emitted code crashes everytime (even on different machine) when `throw` statement is executed.

## What you expected to happen

In a compiled code to be able to use `throw` statement without crash.

## Step-by-step reproduction instructions (by running `brew install` commands)

brew install gcc
compile this file `g++-10 file.cpp`:
```
#include <cstdio>

int main() {
	try {
		throw int{42};
	} catch (int) {
		puts("whops");
	}
}
```
./a.out

it should print `whops` and instead it gives `abort`

**Comment by hanickadot** [comment]
I tried both the bootle and build. Neither of them generates correct code on two distinct machines (both Big Sur and intel).

**Comment by beached** [comment]
same with gcc version 10.2.0 (Homebrew GCC 10.2.0_1) and 
```#include <cstdio>
struct E{};
int main() {
	try {
		throw E{};
	} catch (E const &) {
		puts("whops");
	}
}```

**Comment by carlocab** [comment]
Are you able to reproduce this with a _non_-Homebrew gcc 10.2.0? I think it's only a Homebrew issue if you aren't.

**Comment by hanickadot** [comment]
Ok, it's a bug in GCC, filed here: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98805

**Comment by hanickadot** [comment]
Seems difference that 10.2 was build for Catalina, but run on Big Sur 🤔

**Comment by carlocab** [comment]
Yea, gcc seems to really not like Big Sur a lot. Can you see if you can install the Catalina bottle of `10.2.0_2`, and see if that changes anything?

Just delete the `:big_sur` bottle line from the formula and do `HOMEBREW_NO_AUTO_UPDATE=1 brew reinstall gcc`.

**Comment by hanickadot** [comment]
Yes, the catalina bottle is working on bigsur as it should.

**Comment by carlocab** [comment]
Weird bug. Thanks for the rapid upstream report.

**Comment by hanickadot** [comment]
Yeah, seems so significant I couldn't believe it would be upstream error.

**Comment by alebcay** [comment]
Issue should be patched via https://github.com/Homebrew/homebrew-core/pull/69685.

**Comment by carlocab** [comment]
>Yeah, seems so significant I couldn't believe it would be upstream error.

It might be hard to believe, but bugs this bad usually aren't our fault. 😅

**Comment by carlocab** [comment]
Closed in fd3ae4c644737c54702131514a4c68e6337160f0.

**Comment by hanickadot** [comment]
> > Yeah, seems so significant I couldn't believe it would be upstream error.
> 
> It might be hard to believe, but bugs this bad usually aren't our fault. 😅

Sorry that must sound harsh. Thank you for all awesome work you do with homebrew.
